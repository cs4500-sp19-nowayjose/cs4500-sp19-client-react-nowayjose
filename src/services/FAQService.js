export default class FAQService {
    static instance = null;
    static getInstance() {
        if(FAQService.instance === null) {
            FAQService.instance = new FAQService()
        }
        return this.instance
    }
   
    findFAQById = id =>
        fetch(`https://cs4500-sp19-nowayjose.herokuapp.com/api/faqs/${id}`)
            .then(response => response.json())
    findAllFAQs = () =>
        fetch("https://cs4500-sp19-nowayjose.herokuapp.com/api/faqs")
            .then(response => response.json())
}