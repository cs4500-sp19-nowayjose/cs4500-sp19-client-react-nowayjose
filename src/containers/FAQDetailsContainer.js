import React from 'react'
import FAQService from '../services/FAQService'
import FAQDetails from '../components/FAQDetails'

class FAQDetailsContainer extends React.Component {
    constructor(props) {
        super(props)
        this.faqService = FAQService.getInstance()
        this.state = {
            faqs: [],
            faq: {
                choiceAnswer: '',
                id: 1
            }
        }
    }
    componentDidMount() {
        this.faqService
            .findAllFAQs()
            .then(faqs => {
                this.faqService.findFAQById(this.props.match.params.id).then(faq => {
                    this.setState({
                        faqs: faqs,
                        faq: faq
                    })
                })
            }
            )
    }

    selectFAQ = id =>
        this.faqService
            .findFAQById(id)
            .then(faq => {
                this.props.history.push("/admin/faqs/" + id)
                this.setState({
                    faq: faq
                })
            }
            )
            
    render() {
        return (
            <FAQDetails
                faqs={this.state.faqs}
                faq={this.state.faq}
                selectFAQ={this.selectFAQ}
            />
        )
    }
}

export default FAQDetailsContainer