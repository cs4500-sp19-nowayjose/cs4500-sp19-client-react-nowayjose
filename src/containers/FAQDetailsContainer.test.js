import React from 'react'
import FAQDetailsContainer from './FAQDetailsContainer'
import TestRenderer from 'react-test-renderer';
import FaqsService from '../services/FAQService'
const faqsService = FaqsService.getInstance()
import '../services/FAQService.mock'

test('[FAQDetailsContainer renders correctly]', () => {
    const testRenderer = TestRenderer.create(
        <FAQDetailsContainer/>)
    let tree = testRenderer.toJSON()
    expect(tree).toMatchSnapshot()

    tree = testRenderer.toJSON()
    expect(tree).toMatchSnapshot()
})

test('[FAQDetailsContainer renders correctly after selecting different faq]', () => {
    const testRenderer = TestRenderer.create(
        <FAQDetailsContainer/>)
    let tree = testRenderer.toJSON()
    expect(tree).toMatchSnapshot()

    const testInstance   = testRenderer.root
    let   selectFaq    = testInstance.findByProps({className: 'selectedFaq'})

    selectFaq.props.onChange({target: {value: '4'}})
    tree = testRenderer.toJSON()
    expect(tree).toMatchSnapshot()
})