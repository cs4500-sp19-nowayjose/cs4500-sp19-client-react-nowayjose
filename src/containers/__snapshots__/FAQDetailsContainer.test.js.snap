// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[FAQDetailsContainer renders correctly after selecting different faq] 1`] = `
<div>
  <h3>
    FAQ Details
  </h3>
  <select
    className="selectedFaq"
    onChange={[Function]}
    value={1}
  />
  <br />
  <input
    className="form-control"
    onChange={[Function]}
  />
  <br />
  <input
    className="form-control"
    onChange={[Function]}
  />
</div>
`;

exports[`[FAQDetailsContainer renders correctly after selecting different faq] 2`] = `
<div>
  <h3>
    FAQ Details
  </h3>
  <select
    className="selectedFaq"
    onChange={[Function]}
    value={1}
  />
  <br />
  <input
    className="form-control"
    onChange={[Function]}
  />
  <br />
  <input
    className="form-control"
    onChange={[Function]}
  />
</div>
`;

exports[`[FAQDetailsContainer renders correctly] 1`] = `
<div>
  <h3>
    FAQ Details
  </h3>
  <select
    className="selectedFaq"
    onChange={[Function]}
    value={1}
  />
  <br />
  <input
    className="form-control"
    onChange={[Function]}
  />
  <br />
  <input
    className="form-control"
    onChange={[Function]}
  />
</div>
`;

exports[`[FAQDetailsContainer renders correctly] 2`] = `
<div>
  <h3>
    FAQ Details
  </h3>
  <select
    className="selectedFaq"
    onChange={[Function]}
    value={1}
  />
  <br />
  <input
    className="form-control"
    onChange={[Function]}
  />
  <br />
  <input
    className="form-control"
    onChange={[Function]}
  />
</div>
`;
